src/resources/Api.ts

```diff
@@ -40,14 +41,30 @@ export class Api {
   compile() {
     const resources: CfnResources = {};
 
+    if (this.isExistingApi()) {
+      log.info(`
+          Updating an existing Graphql API.
+          The following configuration options are ignored:
+            - name
+            - authentication
+            - additionalAuthentications
+            - schema
+            - domain
+            - apiKeys
+            - xrayEnabled
+            - logging
+            - waf
+            - tags
+        `);
+    }
```

Comment:
I'd just validate this.
i.e. don't allow these fields is present in the config along with `apiId`
The validator will take care of showing the warning/error (depending on configuration)